### PR TITLE
Krp 282/Account opening request

### PR DESCRIPTION
**Pull request by lera**

### CODE REVIEW

File: src/routes/accountOpeningRequest.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    rejection_reason: {
+      failed_validation: null,
+      details: null,
+    },
+  };
+
+  person.accountOpeningRequests = person.accountOpeningRequests || [];
+  person.accountOpeningRequests.push(accountOpeningRequest);
+
+  await savePerson(person);
```
Comment (UgurGumushan) [inline src/routes/accountOpeningRequest.ts]: We can secure it from thread races using redlock. (I'll create a PR targeting this one)
We would eventually replace the deprecated createPerson flow with this one and we would have issues when e2e user setup uses this function.

File: src/routes/accountOpeningRequest.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  };
+
+  person.accountOpeningRequests = [
+    ...person.accountOpeningRequests.filter(
+      (request) => request.id !== accountOpeningRequest.id
+    ),
+    completedRequest,
+  ];
+
+  await savePerson(person);
```
Comment (UgurGumushan) [inline src/routes/accountOpeningRequest.ts:74]: This saves person without reloading it and would overwrite the previously saved person.

Comment (UgurGumushan) [inline src/routes/accountOpeningRequest.ts]: Person object here doesn't have changes made by createAccount, and would remove account from person

Comment (lera) [reply src/routes/accountOpeningRequest.ts]: it didn't remove it when I tested it, but I see your point. I will update it

Comment (UgurGumushan) [reply src/routes/accountOpeningRequest.ts]: I have a PR which includes this change: https://github.com/kontist/mock-solaris/pull/168

File: src/routes/accounts.ts [hunk 3]
```diff
@@ -117,21 +118,15 @@ export const showPersonAccounts = async (req, res) => {
   res.status(200).send(accounts);
 };
 
-let counter = 0;
-
-export const createAccount = async (personId, data) => {
```
Comment (UgurGumushan) [inline src/routes/accounts.ts:122]: This breaks usages of on the backend. Some code imports MockSolaris functions directly, e.g. import MockSolaris from "@kontist/mock-solaris";
And invoke these functions. You can search "MockSolaris.createAccount(" on the backend project to usages of data property.

Comment (lera) [reply src/routes/accounts.ts:122]: oh... didn't notice it! thanks!

**Review by UgurGumushan — APPROVED** [review]

**Pull request merged**